packages/features/insights/server/routing-events.ts

```diff
@@ -635,6 +645,483 @@ class RoutingEventsInsights {
       return flatResponse;
     });
   }
+
+  static async routedToPerPeriod({
+    teamId,
+    isAll,
+    organizationId,
+    routingFormId,
+    startDate: _startDate,
+    endDate: _endDate,
+    period,
+    cursor,
+    userCursor,
+    limit = 10,
+    searchQuery,
+  }: RoutingFormInsightsTeamFilter & {
+    period: "perDay" | "perWeek" | "perMonth";
+    startDate: string;
+    endDate: string;
+    cursor?: string;
+    userCursor?: number;
+    limit?: number;
+    searchQuery?: string;
+  }) {
+    const dayJsPeriodMap = {
+      perDay: "day",
+      perWeek: "week",
+      perMonth: "month",
+    } as const;
+
+    const dayjsPeriod = dayJsPeriodMap[period];
+    const startDate = dayjs(_startDate).startOf(dayjsPeriod).toDate();
+    const endDate = dayjs(_endDate).endOf(dayjsPeriod).toDate();
+
+    // Build the team conditions for the WHERE clause
+    const formsWhereCondition = await this.getWhereForTeamOrAllTeams({
+      teamId,
+      isAll,
+      organizationId,
+      routingFormId,
+    });
+
+    const teamConditions = [];
+
+    // @ts-expect-error it does exist but TS isn't smart enough when it's number or int filter
+    if (formsWhereCondition.teamId?.in) {
+      // @ts-expect-error same as above
+      teamConditions.push(`f."teamId" IN (${formsWhereCondition.teamId.in.join(",")})`);
+    }
+    if (routingFormId) {
+      teamConditions.push(`f.id = '${routingFormId}'`);
+    }
```

Comment:
Build the conditions to make sure to bring back the selected TeamIDs and routing forms in raw sql.